### DISCUSSION TITLE
Is it possible to load a DynamicData from a string payload (json)?

### DISCUSSION THREAD

**Discussion by cedricmelancon** [eProsima/Fast-DDS#3728] [Ideas]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

The new DynamicDataHelper is adding a way to read a DynamicData in a stringstream. Is it possible to have the inverse behavior, which is to load the content of a DynamicData from a string (like a Json or other format)? This would be used for a publisher from one protocol to another, but I don't want to import all the types.

### Current behavior

N/A

### Steps to reproduce

N/A

### Fast DDS version/commit

2.11.x

### Platform/Architecture

Ubuntu Focal 20.04 amd64, Ubuntu Focal 20.04 arm64

### Transport layer

Default configuration, UDPv4 & SHM

**Comment by JLBuenoLopez** [comment]
Thanks for your contribution @cedricmelancon 

I am labeling this as an enhancement as it is a feature request and moving it to the proper discussion forum as per [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md). You might consider contacting [Fast DDS commercial support team](https://github.com/eProsima/Fast-DDS#getting-help) in order to discuss options about implementing this feature.

**Reply by com-server-ap** [reply]
I also need this feature and look forward to an interface being provided soon.

**Reply by juanlofer-eprosima** [reply]
@cedricmelancon @com-server-ap The requested feature is now available in [Fast DDS v3.3.0](https://github.com/eProsima/Fast-DDS/releases/tag/v3.3.0). Please have a look at the corresponding [documentation section](https://fast-dds.docs.eprosima.com/en/v3.3.0/fastdds/xtypes/type_serializing.html#json-to-dynamicdata) for more information.

**Reply by ploe24** [reply]
> @cedricmelancon @com-server-ap The requested feature is now available in [Fast DDS v3.3.0](https://github.com/eProsima/Fast-DDS/releases/tag/v3.3.0). Please have a look at the corresponding [documentation section](https://fast-dds.docs.eprosima.com/en/v3.3.0/fastdds/xtypes/type_serializing.html#json-to-dynamicdata) for more information.

@juanlofer-eprosima Hi, this is nice, but is there also a way to achieve the same with participant profiles? Like creating a participant from XML/JSON with `load_from_profile("foo.xml")` but with a serialized JSON/XML string instead.

Haven't found it so far :( Would be awesome if you could direct me to where it's mentioned or guide me how to do this (if possible) 

Thanks in advance :))

**Reply by MiguelCompany** [reply]
@ploe24 You have [this method](https://github.com/eProsima/Fast-DDS/blob/master/include/fastdds/dds/domain/DomainParticipantFactory.hpp#L359) in `DomainParticipantFactory`

**Reply by ploe24** [reply]
> @ploe24 You have [this method](https://github.com/eProsima/Fast-DDS/blob/master/include/fastdds/dds/domain/DomainParticipantFactory.hpp#L359) in `DomainParticipantFactory`

Thanks a lot. I must have missed that :)
That's exactly what I was looking for
